### PR TITLE
Simplify NodeGetAttributeTypeExtension

### PR DESCRIPTION
**Pull request by staabm**
remove unnecessary use of `ParametersAcceptorSelector::selectSingle` which is deprecated with PHPStan 2.x

### CODE REVIEW

File: src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php [hunk 1]
```diff
@@ -51,18 +51,16 @@ public function getTypeFromMethodCall(
         MethodReflection $methodReflection,
         MethodCall $methodCall,
         Scope $scope
-    ): Type {
-        $returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
```
Comment (samsonasik) [inline src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: I think to keep 1 - 1 behaviour, `selectFromArgs()` should be used

```php
$returnType = ParametersAcceptorSelector::selectFromArgs($scope, $methodCall->getArgs(), $methodReflection->getVariants())->getReturnType();
```

Comment (staabm) [reply src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: when returning `null` PHPStan will use:

https://github.com/phpstan/phpstan-src/blob/e95f79b2a3fff1f749cdd6dc9eaab2d2600def1a/src/Analyser/MutatingScope.php#L5616-L5621

Comment (samsonasik) [reply src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: `getReturnType()` return PHPStan type afaik, it maybe missing test here, but allow return null is changing behavior here

Comment (staabm) [reply src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: see https://phpstan.org/developing-extensions/dynamic-return-type-extensions

grafik

Comment (TomasVotruba) [reply src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: If I understand it correctly, it fallbacks to `selectFromArgs()` when `null` is returned.

LGTM :+1:

**Comment by TomasVotruba** [comment]
Thanks :+1:

File: src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php (hunk 1 above)
Comment (staabm) [reply src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php:55]: exactly
